### ISSUE TITLE
dnd-kit adapter keyboard nav does not work

### ISSUE THREAD

**Issue by mcmcgrath13** [react-querybuilder/react-querybuilder#1031]
### React Query Builder version

v8.15.0

### Platform

Firefox on M1 Mac

### Description

The Keyboard navigation does not work with the dnd-kit adapter. The spacebar, then arrow keys do not do anything.

It looks like the `KeyboardSensor` needs all of the event and not just the `nativeEvent`, so changing 

```ts
      const nativeHandler: EventListener = e => handler({ nativeEvent: e });
```

to

```ts
      const nativeHandler: EventListener = e => handler({ ...e, nativeEvent: e });
```
should likely make this work. Additionally, I wonder if adding the Sortable context would improve keyboard nav.

I am starting a branch to work on these and am happy to contribute back if you think this is the correct fix

### Reproduction

Follow example in documentation for using the dndkit adapter, try to use keyboard nav (spacebar on drag handle, then arrow keys) and note nothing happens

### Expected behavior

hitting space on drag handle should enable arrow keys to move items around

**Comment by jakeboone02** [comment]
Thanks for the report. Feel free to submit a PR with whatever fix you think is appropriate!

**Comment by mcmcgrath13** [comment]
I opened up #1032 to address the main problem of it not working at all.

I'm playing around with improving the keyboard nav experience in https://github.com/mcmcgrath13/react-querybuilder/pull/1

It largely makes things smoother, but there are some funky edges on occasion, so I'm a little on the fence about whether it should be the default. Another option would be to make the coordinate getter an optional option to the `createDndKitAdapter` function so people _can_ mess around with this. Thoughts on how to proceed? (or any ideas on why the up/down arrows occasionally won't move smoothly?)

**Comment by jakeboone02** [comment]
I'm not really in love with any of RQB's drag-and-drop code, so go with your gut when it comes to proposing stuff&mdash;especially non-breaking changes like additional config options. The library is small and modular enough that I'm not overly concerned about bundle size, API surface area, etc. Write it how you want it to work and we'll go from there.

RQB gets very few external PRs, and I tend to accept them as-is if I'm either not intimately familiar with the topic or don't want to/can't spend the time to optimize the implementation. Drag-and-drop falls in category, mostly for the latter reason. (I guess that's probably a subtle way of answering your last question about why it doesn't work smoothly sometimes! 😄)

**Comment by jakeboone02** [comment]
Your fix for this in #1032 was included in [v8.16.1](https://github.com/react-querybuilder/react-querybuilder/releases/tag/v8.16.1).

I also added you as a contributor: 7d01c9d76d204f3955e15e4d7800be7aeb0b08ce

Thanks!
